tests/test_events/test_broker.py

```diff
@@ -68,3 +69,15 @@ def test_broker_continues_even_if_one_handler_failed(self, caplog: LogCaptureFix
                 "Handling event failed, event: Event(type='x', data={'y': 1})",
             )
         ]
+
+    def test_broker_events_are_not_affected_by_specific_event_handler(self) -> None:
+        def handler(event: Event) -> None:
+            expected_event = Event({"y": 1}, event_type="x")
+            assert event == expected_event
```

Comment:
In general, this test is not bad but each error raised directly in the handler is caught and logged.

Try to run it without `deepcopy()` - the test will pass anyway 😉